sources/api/pluto/src/main.rs

```diff
@@ -269,9 +309,23 @@ async fn run() -> Result<()> {
     let setting_name = parse_args(env::args());
     let mut client = ImdsClient::new().await.context(error::ImdsClient)?;
 
+    // Retrieve the kubernetes network configuration for the EKS cluster
+    let aws_k8s_info = api::get_aws_k8s_info().await.context(error::AwsInfo)?;
+    let cluster_network_config =
+        eks::get_cluster_network_config(&aws_k8s_info.region, &aws_k8s_info.cluster_name)
+            .await
+            .context(error::EksError)?;
```

Comment:
This results in us always making a `DescribeCluster` request for each `pluto` invocation, even if we don't need the response to determine `cluster-dns-ip` or `node-ip`.

There are a few drawbacks to that: it's slow and fallible; it creates unnecessary load on EKS; and the node might not have the role needed to describe the cluster.

It seems like we should first calculate `cluster-dns-ip`, if we don't have it in settings, and then either we're done (if we were asked to generate it) or we can pass the final value into `get_node_ip` (which would also simplify its interface).

Ideally we'd have a way to ensure that we generate `cluster-dns-ip` first before we attempt to generate `node-ip`, so we'd only need to describe the cluster once - not sure that's feasible with the tools we have right now, and it's not a blocker.